src/sentry/static/sentry/app/components/confirm.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-bootstrap/lib/Modal';
+
+import Button from './buttons/button';
+
+class Confirm extends React.PureComponent {
+  static propTypes = {
+    disabled: PropTypes.bool,
+    message: PropTypes.string.isRequired,
+    onConfirm: PropTypes.func.isRequired,
+    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
+    confirmText: PropTypes.string.isRequired,
+    cancelText: PropTypes.string.isRequired
+  };
+
+  static defaultProps = {
+    priority: 'primary',
+    cancelText: 'Cancel',
+    confirmText: 'Confirm'
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {
+      isModalOpen: false
+    };
+  }
+
+  handleConfirm = () => {
+    this.setState({
+      isModalOpen: false
+    });
+
+    this.props.onConfirm();
```

Comment:
what about protecting from double triggering here? if you think that's reasonable. if you don't think the risk is worth the delay i trust that opinion

Reply:
I think currently we use `confirm` for destructive API calls, but since we can't predict how this will be used in the future it's a good idea to add this. It shouldn't cause a delay.

Reply:
i guess i meant delay in shipping this improvement, not like debouncing :P but it sounds like i meant debouncing